core/domain/auth_domain.py

```diff
@@ -181,9 +181,10 @@ def to_dict(self):
             dict(str: *). A dict with UserAuthDetailsModel property names as
             keys and values from self.
         """
-        return {
-            'gae_id': self.gae_id,
-            'firebase_auth_id': self.firebase_auth_id,
-            'parent_user_id': self.parent_user_id,
-            'deleted': self.deleted
-        }
+        property_values = [
```

Comment:
I don't get why this was changed. Let's please avoid introducing one-off conventions for methods that are common across the codebase. If you want to do so, please do the refactoring as part of its own PR instead unless there is already a design/refactoring doc and general written agreement that this is the format we want to move towards, otherwise it will confuse developers who aren't sure which convention to follow.

Reply:
I did it this way because I don't want `None` values to overwrite previous values, not because I want to introduce a new convention.

Reply:
But why does this consideration not apply to other domain objects?

Reply:
This is also a compromise I made for handling the split-responsibility over `UserAuthDetailsModel`, let's discuss if it's worthwhile there: https://github.com/oppia/oppia/pull/11588#discussion_r554284761.

Reply:
I don't understand how this relates to the discussion there, sorry. Could you please elaborate?

I.e. what exactly will happen if we did it the usual way? Normally to_dict() surfaces everything so at the very least we shouldn't be naming it to_dict().

Reply:
OK, I've given it a shot. WDYT?

Reply:
Obsolete.